django/db/models/base.py

```diff
@@ -114,8 +115,14 @@ def __new__(cls, name, bases, attrs):
                     app_label_index = package_components.index(MODELS_MODULE_NAME) + 1
                 except ValueError:
                     app_label_index = 1
-                kwargs = {"app_label": package_components[app_label_index]}
-
+                try:
+                    kwargs = {"app_label": package_components[app_label_index]}
+                except IndexError:
+                    raise ImproperlyConfigured((
+                        'Unable to detect the app label for model '
```

Comment:
line breaking is a bit odd -- %s could fit on the previous line
also, one space after period